src/Extension/CoreExtension.php

```diff
@@ -686,12 +686,12 @@ public static function merge(...$arrays): array
     /**
      * Slices a variable.
      *
-     * @param mixed $item         A variable
-     * @param int   $start        Start of the slice
-     * @param int   $length       Size of the slice
-     * @param bool  $preserveKeys Whether to preserve key or not (when the input is an array)
+     * @param \Traversable|array|string $item         A variable
+     * @param int                       $start        Start of the slice
+     * @param int                       $length       Size of the slice
+     * @param bool                      $preserveKeys Whether to preserve key or not (when the input is an array)
      *
-     * @return mixed The sliced variable
+     * @return ($item is string ? string : array)
```

Comment:
Any other method that would benefit from this change?

Reply:
Can we merge this PR for now? I'm working occasionally on TwigStan and will propose new PR's when I find new things.